Docs: OaC/Git Sync updates for Obs Con (#111792)
grafana/grafana: docs/sources/observability-as-code/provision-resources/git-sync-setup.md

# Set up Git Sync

{{< admonition type="caution" >}}

> TBC

Reviewer: Let's not forget to update this

| Adds a table summarizing changes to your pull request | Provides a convenient way to save changes back to GitHub.                       | Webhooks configured                    |
| Add a dashboard preview image to a PR                 | View a snapshot of dashboard changes to a pull request without opening Grafana. | Image renderer and webhooks configured |

{{< admonition type="note" >}}

Alternatively, you can configure a local file system instead of using GitHub. Refer to [Set up file provisioning](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/observability-as-code/provision-resources/file-path-setup/) for more information.

{{< /admonition >}}

## Performance impacts of enabling Git Sync

Git Sync is an experimental feature and is under continuous development. Reporting any issues you encounter can help us improve Git Sync.

When Git Sync is enabled, the database load might increase, especially for instances with a lot of folders and nested folders. Evaluate the performance impact, if any, in a non-production environment.

## Before you begin

{{< admonition type="caution" >}}

> See also [Known limitations](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/observability-as-code/provision-resources/intro-git-sync#known-limitations/) before using Git Sync.

Reviewer: ```suggestion
Refer to [Known limitations](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/observability-as-code/provision-resources/intro-git-sync#known-limitations/) before using Git Sync.
```

- Enable the required feature toggles in your Grafana instance. Refer to [Enable required feature toggles](#enable-required-feature-toggles) for instructions.
- A GitHub repository to store your dashboards in.
  - If you want to use a local file path, refer to [the local file path guide](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/observability-as-code/provision-resources/file-path-setup/).
- A GitHub access token. The Grafana UI will prompt you during setup.
- Optional: A public Grafana instance.

> - Optional: Image Renderer plugin to save image previews with your PRs.

Reviewer 2: image renderer plugin has been deprecated, worth finding out how it works now

Author: https://github.com/grafana/grafana-image-renderer/issues/782#issuecomment-3328584172 no worries then.
